dissect/target/plugins/os/unix/linux/network.py

```diff
@@ -337,4 +337,135 @@ def _parse_gateway(self, value: str | None) -> NetAddress | None:
         return ip_address(value)
 
 
-MANAGERS = [NetworkManagerConfigParser, SystemdNetworkConfigParser]
+class ProcConfigParser(LinuxNetworkConfigParser):
+    """Parser for dynamic network configuration data from /proc/net.
+
+    Parse gateways, interface names and network from /proc/net/route.
+    Corroborate with TCP connections from /proc/net/tcp to find local IP addresses.
+    Locally bound Ipv6 addresses are parsed from /proc/net/if_inet6.
+    """
+
+    @dataclass
+    class ParserContext:
+        name: str | None = None
+        ip_interfaces: set[NetInterface] = field(default_factory=set)
+        gateways: set[NetAddress] = field(default_factory=set)
+
+        def to_record(self, target: Target) -> UnixInterfaceRecord:
+            return UnixInterfaceRecord(
+                name=self.name,
+                cidr=self.ip_interfaces,
+                gateway=list(self.gateways),
+                source="/proc/net/route",
+                configurator="proc",
+                _target=target,
+            )
+
+    def interfaces(self) -> Iterator:
+        interfaces: dict[str, ProcConfigParser.ParserContext] = {}
+
+        routes = self._parse_proc_net_route(interfaces)
+
+        # Find locally bound IPv4 addresses from /proc/net/tcp and correlate with routes
+        tcp_local_ipv4 = self._parse_proc_net_tcp_local_ipv4()
```

Comment:
Reviewer: we could also / alternatively parse `/proc/net/fib_trie` 

```
+-- 127.0.0.0/8 2 0 2
        +-- 127.0.0.0/31 1 0 0
           |-- 127.0.0.0
              /8 host LOCAL
           |-- 127.0.0.1
              /32 host LOCAL
        |-- 127.255.255.255
           /32 link BROADCAST
```

And match the /32 LOCAL entries, since the entries in tcp are ephemeral.
(but `fib_trie` is a bit dodgy to parse)